optimisation/design/Third_Party_API_Design.md

```diff
@@ -0,0 +1,595 @@
+# An Application Programming Interface for user defined model functions in PACE
+**2020-05-15**
+
+
+# Scope
+
+This documents details the required function signatures for user defined sample scattering models
+and how they interface with Horace, using either Matlab, Python, C, C++ or Fortran.
+
+
+# Motivation
+
+Analysis of inelastic neutron scattering data almost always requires a parameterised model of the measured scattering intensities.
+Horace, as currently implemented, requires this model to be in the form of a Matlab function.
+This function is required to either 
+
+1. Take a 4D vector of momentum-energy ![\mathbb{Q} \equiv (\mathb{Q},E)] and yield the scattering function ![S(\mathbb{Q})].
+2. Take the 3D vector of momentum transfer ![\mathbf{Q}] only and yield ![N] sets of vectors ![(E_n, S(\mathbf{Q}, E_n))] where ![N] is the number of dispersion branches.
```

Comment:
I would propose a third variant which takes **Q** and returns E<sub>n</sub>, S(**Q**,E<sub>n</sub>), and γ(**Q**,E<sub>n</sub>); where γ(**Q**,E<sub>n</sub>) *is* the intrinsic energy width of the n<sup>th</sup> mode at **Q**.

Reply:
This text is about the current implementation...